Procurement/Utility/ClientLogFileWatcher.cs

```diff
@@ -0,0 +1,191 @@
+﻿using POEApi.Infrastructure;
+using POEApi.Model;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Procurement.Utility
+{
+    public class ClientLogFileEventArgs : EventArgs
+    {
+        public DateTime EventDateTime { get; private set; }
+        public long EventTimestamp { get; private set; }
+        public string LocationEntered { get; private set; }
+
+        public ClientLogFileEventArgs(DateTime eventDateTime, long eventTimestamp, string locationEntered)
+        {
+            EventDateTime = eventDateTime;
+            EventTimestamp = eventTimestamp;
+            LocationEntered = locationEntered;
+        }
+    }
+
+    class ClientLogFileWatcher
+    {
+        private static ClientLogFileWatcher _instance;
+        public static ClientLogFileWatcher Instance
+        {
+            get
+            {
+                if (_instance == null)
+                    _instance = new ClientLogFileWatcher();
+
+                return _instance;
+            }
+        }
+
+        protected static System.IO.FileSystemWatcher FileWatcher
+        {
+            get;
+            private set;
+        }
+
+        public delegate void ClientLogFileEventHandler(ClientLogFileWatcher sender, ClientLogFileEventArgs e);
+        public static event ClientLogFileEventHandler ClientLogFileChanged;
+
+        public DateTime LastDateTimeSeen
+        {
+            get;
+            protected set;
+        }
+
+        public long LastTimestampSeen
+        {
+            get;
+            protected set;
+        }
+
+        public long LastFileSizeSeen
+        {
+            get;
+            protected set;
+        }
+
+        protected System.Timers.Timer PollingTimer
+        {
+            get;
+            set;
+        }
+
+        protected Regex LocationChangedRegex
+        {
+            get;
+            set;
+        }
+
+        protected void Initialize()
+        {
+            if (!Settings.UserSettings.Keys.Contains("ClientLogFileLocation"))
+                return;
+            string fullFilePath = Settings.UserSettings["ClientLogFileLocation"];
+            if (string.IsNullOrWhiteSpace(fullFilePath))
+                return;
+
+            FileWatcher = new System.IO.FileSystemWatcher();
+            FileWatcher.Path = System.IO.Path.GetDirectoryName(fullFilePath);
+            FileWatcher.Filter = System.IO.Path.GetFileName(fullFilePath);
+            FileWatcher.NotifyFilter = NotifyFilters.LastWrite | NotifyFilters.Size | NotifyFilters.LastAccess;
+
+            FileWatcher.Changed += OnFileChanged;
+
+            PollingTimer = new System.Timers.Timer();
+            PollingTimer.Elapsed += (s, e) => { ReadClientLogFile(); };
```

Comment:
Having an anonymous signature like this means you will be unable to unsubscribe from the event later on.

Reply:
As per my other comment, I think this is ok, since the subscriber and publisher's lifetimes are the same, and does not need to be unsubscribed.  Please correct me if I am wrong.